collector/zfs_linux.go

```diff
@@ -47,6 +47,23 @@ func (c *zfsCollector) updateZfsStats(subsystem string, ch chan<- prometheus.Met
 	})
 }
 
+func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+	zpoolPaths, err := filepath.Glob(procFilePath(filepath.Join(c.linuxProcpathBase, c.linuxZpoolIoPath)))
+	if err != nil {
+		return err
+	}
+
+	if zpoolPaths == nil {
+		return err
```

Comment:
Since you already checked the error, just `return nil` to make it more obvious that there's nothing to be done.